### PR TITLE
CommandPalette: Populate dashboard search when the palette is opened

### PR DESCRIPTION
**Pull request by ryantxu**
This PR makes a few changes for search in the new Command palette:
1. It uses the `getGrafanaSearcher()` API, rather than backendSrv -- this will delegate search to whatever is configured
2. It waits until the command palette is opened to execute the search query
3. changes the formatting so it does not duplicate the "Go to dashboard" text

Now:
| Before | After |
|---|--|
| image | image |

I am VERY excited to finally know what was calling the search endpoint all the time :) 

Down the road, we should likely look at how this can/should pass the typed text to the search query, but I think this is fine for now

### CODE REVIEW

File: public/app/features/commandPalette/actions/dashboard.nav.actions.ts [hunk 1]
```diff
@@ -1,21 +1,24 @@
...
-      parent: parentId,
-      id: `go/dashboard/${item.url}`,
-      name: `Go to dashboard ${item.title}`,
-      perform: () => {
-        locationService.push(locationUtil.stripBaseFromUrl(item.url));
-      },
-    }));
+  const goToDashboardActions: Action[] = data.view.map((item) => ({
+    parent: parentId,
+    id: `go/dashboard/${item.url}`,
```
Comment (ryantxu) [inline public/app/features/commandPalette/actions/dashboard.nav.actions.ts:16]: does this just need to be unique?  if yes, let's use
```suggestion
    id: `go/dashboard/${item.uid}`,
```

Comment (gelicia) [reply public/app/features/commandPalette/actions/dashboard.nav.actions.ts:16]: Hmm, we use this partially for the usage tracking as well. Having the URL was something a bit more readable. I'll just track the ID and name if we ever want to dive in a little - either that or we find out it's worthless to track something specific to instances and we just group the go to dashboard actions as one thing.

**Review by gelicia — CHANGES_REQUESTED** [review]
I'm not sure exactly whats happening here, but the dashboard I select isn't the one it ends up navigating to. I put a console log out in the perform and it's not like the name is one thing and the URL is another - its selecting a different action. 

https://user-images.githubusercontent.com/1533128/175297492-50fb6cc5-e1fe-40a8-872c-82a3fc315abc.mov

File: public/app/features/commandPalette/actions/dashboard.nav.actions.ts (hunk 1 above)
Comment (ryantxu) [reply public/app/features/commandPalette/actions/dashboard.nav.actions.ts:16]: makes sense

**Comment by gelicia** [comment]
@ryantxu wondering what you think about some sort of loading state for this. Basically what I want to do is remove as much as possible out of that first `useEffect` since we load and populate the actions later ([src](https://github.com/grafana/grafana/blob/main/public/app/features/commandPalette/CommandPalette.tsx#L51)).

If you remove that completely, the command palette doesn't work its animated keyframes correctly because of a timing issue. Even if that were not the case, if there would be a delay because of the dashboard search loading, I think it would be good to communicate why that is happening. The video below has my network throttled to show what I mean more clearly. cc: @Elfo404 
https://user-images.githubusercontent.com/1533128/175399372-22b7e1e7-123a-4283-9ec8-cb97fd4df05d.mov

**Comment by ryantxu** [comment]
@gelicia -- nice.  I pushed a variation on your suggestion.  It will load the static things first, then add in the dashboards however quickly the response returns

**Comment by gelicia** [comment]
@ryantxu Sorry about the uno reverse card here, but I changed up the hooks a little bit for the Command Palette, can you let me know what you think?

File: public/app/features/commandPalette/CommandPalette.tsx [hunk 2]
```diff
@@ -48,18 +49,22 @@ export const CommandPalette = () => {
...
-        const staticActions = getGlobalActions(navBarTree);
-        const dashAct = await getDashboardNavActions('go/dashboard');
-        setActions([...staticActions, ...dashAct]);
+        setStaticActions(getGlobalActions(navBarTree));
+        setActions(staticActions);
       }
     })();
     // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, [isNotLogin]);
+  }, [isNotLogin, navBarTree]);
```
Comment (gelicia) [inline public/app/features/commandPalette/CommandPalette.tsx:57]: Adding this here to future proof it, if navBarTree is ever something that could change dynamically

**Comment by ryantxu** [comment]
Looks good!  thanks

**Review by gelicia — APPROVED** [review]

**Pull request merged**